### PR TITLE
discussion/spike: Model update proposal

### PR DESCRIPTION
**Pull request by kraman**
Summary of changes:
* Separate ServiceInstance into ServiceContainer (stored in redis) and ServiceInstance (stored in PM-memory)
* Hide update/delete methods for read-only models
* Rename ServerService -> Service
* Rename InstanceAction -> Action
* Convert IDs to be GUID strings instead of numeric

Unimplemented ideas:
* Move to remote-methods like /instance/stop?soft=true instead of a soft-stop Action

### CODE REVIEW

File: models/metric.txt [hunk 1]
```diff
@@ -1,66 +0,0 @@
-service/server-service:
```
Comment (sam-github) [inline models/metric.txt]: should be a seperate commit, its unrelated

File: models/executor.js [hunk 2]
```diff
@@ -5,4 +5,28 @@ module.exports = function(Executor) {
     }
     next();
   };
+
+  Executor.disableRemoteMethod('__delete__instances');
```
Comment (sam-github) [inline models/executor.js]: why?

Comment (sam-github) [inline models/executor.js]: I mean, why are any of these being disabled? We have REST API, they use it, what problem are we solving?

File: models/metric.txt (hunk 1 above)
Comment (sam-github) [inline models/metric.txt]: also, this file was my attempt at understanding the relationships between the models

IMO, @altsang @ritch @seanbrookes @raymondfeng Loopback/strong-arc ***desperately*** needs a model relationship viewer that graphically displays the relationships between models. Figuring it out from the json is pretty hard, I always end up drawing a picture on paper, or writing a text file summary like this.

File: models/profile-data.js [hunk 3]
```diff
@@ -8,4 +8,10 @@ module.exports = function(ProfileData) {
     this.id = this.id || profileId;
     next();
   };
+
+  ProfileData.disableRemoteMethod('create', true);
```
Comment (sam-github) [inline models/profile-data.js]: I think code like this deserves a comment.

File: models/executor.js (hunk 2 above)
Comment (kraman) [inline models/executor.js]: Some of these models are managed and meant to be created/deleted only by PM or Mesh etc. They should only be queryable and not editable over the REST API.

File: models/metric.txt (hunk 1 above)
Comment (kraman) [inline models/metric.txt]: I'll split this out add a SVG diagram to the repo to explain

File: models/profile-data.json [hunk 4]
```diff
@@ -6,15 +6,16 @@
   "properties": {
     "id": {
       "type": "string",
-      "index": true,
-      "id": true
+      "required": true,
```
Comment (sam-github) [inline models/profile-data.json]: I think only required changed, but the order of the other two fields was switched.

File: models/service-container.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+module.exports = function(ServiceContainer) {
```
Comment (sam-github) [inline models/service-container.js]: ?

**Comment by sam-github** [comment]
I think this would be easier to read if each change was an individual commit, like your point form list in the PR. Its quite difficult to read large changes to json, particularly because its not possible to put comments in json

Other than that, LGTM, though I've concerns about backwards compatibility.

**Pull request closed without merging**